### ISSUE TITLE
Color-code event parameters based on types

### ISSUE THREAD

**Issue by ewpatton** [mit-cml/appinventor-sources#1875]
**Describe the desired feature**

Color-code event parameter entries based on type extracted from Java. If type doesn't map cleanly to a YAIL type (e.g., any) then use the variable orange color as we do today.

**Give an example of how this feature would be used**

There is sometimes lack of clarity about the type of an event parameter. Additional color coding could reveal this information in a more salient way.

**Why doesn't the current App Inventor system address this use case?**

Subtlety of naming a variable might hide its type, or preconceived notions from other languages might have an influence on how people interpret names. There may also be things lost in translation when names are internationalized. Using color would give a non-written cue as to the type of the variable.

**Why is this feature beneficial to App Inventor's educational mission?**

We color code blocks to make it easier to understand their purpose and how they fit with other blocks. This proposal aligns with that motivation.

**Example**

Here is an example of what this might look like for the `BluetoothLE` extension's `StringValuesReceived` block. The `serviceUuid` and `characteristicUuid` parameters are text, so take on the red text color and the `stringValues` parameter is a list so it takes on the list blue color. If the app inventor drags in the block, the color applies to the variable getter block as well iff it resolves to the event parameter and hasn't been set prior to the current use  (at which point we may not be certain of its type any more).

![Color-coded BLE event block](https://user-images.githubusercontent.com/1331253/66451626-4eea9300-ea2b-11e9-82a2-6ce15de2a186.png)

**Comment by ewpatton** [comment]
cc @kdclang @stezelMIT @fturbak @SusanRatiLane @halatmit @jisqyv 

I'm curious what you all think about this proposal.

**Comment by SusanRatiLane** [comment]
Hmm. I'm not sure it makes sense to color code primitive types. We're very fluid about string vs number, etc. However, distinguishing between a primitive type and a data structure like a list or object makes a lot of sense.

**Comment by ewpatton** [comment]
> We're very fluid about string vs number, etc. 

This is true in how we interpret these things, but the value spaces of strings is larger than the value space of numbers, so knowing that a variable is a text versus number is useful. Also, in event handlers we should always be using the most appropriate type, I would consider it a bad design choice if we chose to provide a number as a String in the handler rather than using the appropriate int, long, etc.
